installation-manager/src/main/java/org/wildfly/core/instmgr/InstMgrCreateSnapshotHandler.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.core.instmgr;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationDefinition;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
+import org.jboss.as.controller.registry.OperationEntry;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.installationmanager.MavenOptions;
+import org.wildfly.installationmanager.spi.InstallationManager;
+import org.wildfly.installationmanager.spi.InstallationManagerFactory;
+
+/**
+ * Operation handler that creates a snapshot of the installation-manager configuration.
+ */
+public class InstMgrCreateSnapshotHandler extends InstMgrOperationStepHandler {
+    public static final String OPERATION_NAME = "clone-export";
+
+    public static final OperationDefinition DEFINITION = new SimpleOperationDefinitionBuilder(OPERATION_NAME, InstMgrResolver.RESOLVER)
+            .withFlags(OperationEntry.Flag.HOST_CONTROLLER_ONLY).setRuntimeOnly().build();
+
+    public InstMgrCreateSnapshotHandler(InstMgrService imService, InstallationManagerFactory imf) {
+        super(imService, imf);
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                context.acquireControllerLock();
+                try {
+                    final Path serverHome = imService.getHomeDir();
+                    final Path temporalDir = imService.getControllerTempDir();
+                    final MavenOptions mavenOptions = new MavenOptions(null, false);
+                    final InstallationManager im = imf.create(serverHome, mavenOptions);
+
+                    Path exportFile = temporalDir.resolve("installer-clone-export" + UUID.randomUUID() + ".zip");
+                    addCleanTempFileCompleteStep(context, exportFile);
```

Comment:
It looks like the stream is closed after the complete step, so we should not delete the temporal file here.

We could load the file in memory or we can leave it in the server temporal directory or OS temp dir and delete it when the JVM exits. I'm more inclined toward the latter.

@bstansberry Before changing it, do you agree this is an issue? I assume the stream is closed after the complete steps and it could be even consumed after the operation returns to the client.